pkg/cluster/cluster.go

```diff
@@ -346,21 +320,27 @@ func (c *Cluster) Create() error {
 	//
 	// Do not consider connection pooler as a strict requirement, and if
 	// something fails, report warning
-	roles := c.RolesConnectionPooler()
-	for _, r := range roles {
-		if c.ConnectionPooler != nil {
-			c.logger.Warning("Connection pooler already exists in the cluster")
-			return nil
+	if c.needConnectionPooler() {
+
+		roles := c.RolesConnectionPooler()
+		for _, r := range roles {
+			c.logger.Warningf("found roles are %v", r)
```

Comment:
why `Warningf` and not `Infof` ? at this point there is technically nothing bad happening